test/goal_test.exs

```diff
@@ -1025,6 +1041,287 @@ defmodule GoalTest do
              }
     end
 
+    test "list of primitive types" do
+      data = %{
+        "strings" => [" hello", "world "]
+      }
+
+      schema = %{
+        strings: [
+          type: {:array, :string},
+          rules: [trim: true]
+        ]
+      }
+
+      changeset = Goal.build_changeset(schema, data)
+
+      assert changeset.valid?
+      assert changes_on(changeset) == %{strings: ["hello", "world"]}
+    end
+
+    test "list of incorrect primitive types" do
```

Comment:
The test title suggests it's the primitive types that are wrong but it's actually the values that are incorrect. What about `"list of strings with incorrect values"`?